pkg/credentialprovider/plugin/plugin_test.go

```diff
@@ -418,7 +418,7 @@ func Test_encodeRequest(t *testing.T) {
 			request: &credentialproviderapi.CredentialProviderRequest{
 				Image: "test.registry.io/foobar",
 			},
-			expectedData: []byte(`{"kind":"CredentialProviderRequest","apiVersion":"credentialprovider.kubelet.k8s.io/v1alpha1","image":"test.registry.io/foobar"}
```

Comment:
I think it would be good to keep both tests that validate against  v1alpha1 and v1beta1 versions

Reply:
There is no version check in decode request, added validation in encode against both the versions